### DISCUSSION TITLE
How to set to use swe

### DISCUSSION THREAD

**Discussion by Window-Line** [mozman/ezdxf#677] [Q&A]
How to use swept_surface?
I want to make pipe line using circle(ct) and pathway line(l1).

import ezdxf
from ezdxf.math import OCS, UCS, Vec3
import math
from ezdxf.entities import ExtrudedSurface

doc = ezdxf.new('R2010')
msp = doc.modelspace()

ucs = UCS(origin=(0, 0, 0)).rotate_local_x(math.radians(-60))

l1=msp.add_line(start=(0,3,1), end = (0.4,1),dxfattribs={'color': 1}, ).transform(ucs.matrix)
ct=msp.add_circle(center=(0,3,1), radius=2, dxfattribs={'color': 1}, ).transform(ucs.matrix)

cx=msp.add_swept_surface(dxfattribs={'color': 6})
cx.dxf.swept_entity_id=33
cx.dxf.path_entity_id=32

doc.saveas("meshes2.dxf")

**Comment by mozman** [comment]
You have to create the `SAT` content by yourself set the ACIS code by `entity.acis_data`, `ezdxf` does not provide an ACIS kernel. And `ezdxf` does not support setting `SAB` data which is the format used from R2013 onward. I am afraid this information will not make you happy.

I general you can not create ACIS based entities with `ezdxf`.

**Reply by Window-Line** [reply]
Thank you for your fast feedback. 😊
